### PR TITLE
Feature/farm 443 show farm details

### PR DESCRIPTION
**Pull request by jessicahernandezhv**
[[FARM-443 - Show Farm Details](https://amidodevelopment.atlassian.net/secure/RapidBoard.jspa?rapidView=150&projectKey=FARM&modal=detail&selectedIssue=FARM-443)]

#### 📲 What

Screen to show the user farm details.

#### 🤔 Why
		
Because we need to show a summary of the farmer plot details.
		
#### 🛠 How
		
Using the WAMF Flutter code style.

#### 👀 See
		
farmDetails
		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [x] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/ui/profile/FarmDetailsListItem.dart [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        spacing: _Constants.wrapSpacing,
+        runSpacing: _Constants.wrapRunSpacing,
+        children: <Widget>[
+          _buildCircles(_viewModel.color),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildCircles(Color color) {
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/FarmDetailsListItem.dart]: minor typo.. 
```suggestion
  Widget _buildCircle(Color color) {
```

File: lib/ui/profile/FarmDetails.dart [hunk 2]
```diff
@@ -0,0 +1,210 @@
+import 'package:farmsmart_flutter/ui/common/ListDivider.dart';
+import 'package:farmsmart_flutter/ui/common/roundedButton.dart';
+import 'package:farmsmart_flutter/ui/profile/FarmDetailsListItem.dart';
+import 'package:farmsmart_flutter/ui/common/ContextualAppBar.dart';
+import 'package:farmsmart_flutter/ui/common/ActionSheet.dart';
+import 'package:farmsmart_flutter/ui/common/ActionSheetListItem.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+class _Strings {
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/FarmDetails.dart:11]: 👍

File: lib/ui/profile/FarmDetails.dart [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+  static final BorderRadius buttonRadius = BorderRadius.all(
+    Radius.circular(12),
+  );
+}
+
+class FarmDetailsViewModel {
+  List<FarmDetailsListItemViewModel> items;
+  final String buttonTitle;
+  final Function onConfirmDetails;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/FarmDetails.dart]:
```suggestion
  final Function continue;
```

File: lib/ui/profile/FarmDetailsListItem.dart [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+  static final BorderRadius circleBorderRadius = BorderRadius.all(
+    Radius.circular(30),
+  );
+}
+
+class FarmDetailsListItemViewModel {
+  String title;
+  String detail;
+  Color color;
```
Comment (albertmiro) [inline lib/ui/profile/FarmDetailsListItem.dart]: These values should be final

**Review by LeeMatthewHiggins — COMMENTED** [review]
Looks great just some tiny naming

File: lib/ui/profile/FarmDetailsListItem.dart (hunk 1 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/FarmDetailsListItem.dart]: Done

File: lib/ui/profile/FarmDetails.dart [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+  static final double bottomMargin = 112;
+  static final double buttonHeight = 48;
+
+  static final BorderRadius buttonRadius = BorderRadius.all(
+    Radius.circular(12),
+  );
+}
+
+class FarmDetailsViewModel {
+  List<FarmDetailsListItemViewModel> items;
```
Comment (davidfranquet) [inline lib/ui/profile/FarmDetails.dart]:
```suggestion
  final List<FarmDetailsListItemViewModel> items;
```

Comment (LeeMatthewHiggins) [inline lib/ui/profile/FarmDetails.dart]: that should be final

File: lib/ui/profile/FarmDetails.dart (hunk 3 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/FarmDetails.dart]: Flutter tells me that continue its a keyword

File: lib/ui/profile/FarmDetailsListItem.dart (hunk 4 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/FarmDetailsListItem.dart]: Done

File: lib/ui/profile/FarmDetails.dart [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+      ),
+    ).build(context);
+  }
+
+  void _moreTapped(ActionSheet sheet, BuildContext context) {
+    ActionSheet.present(sheet, context);
+  }
+
+  void _removeAction(FarmDetailsViewModel viewModel, BuildContext context) {
+    viewModel.removeProfile(); //TODO: add the confirm when ready
```
Comment (davidfranquet) [inline lib/ui/profile/FarmDetails.dart]: add the general Alert here

File: lib/ui/profile/FarmDetails.dart (hunk 3 above)
Comment (LeeMatthewHiggins) [reply lib/ui/profile/FarmDetails.dart]: LOL yes it is.. continueAction maybe?

Comment (LeeMatthewHiggins) [reply lib/ui/profile/FarmDetails.dart]: infact just "confirm" will do it

File: lib/ui/profile/FarmDetails.dart (hunk 6 above)
Comment (LeeMatthewHiggins) [reply lib/ui/profile/FarmDetails.dart]: 👍

File: lib/ui/profile/FarmDetails.dart (hunk 5 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/FarmDetails.dart]: Done

File: lib/ui/profile/FarmDetails.dart (hunk 6 above)
Comment (davidfranquet) [reply lib/ui/profile/FarmDetails.dart]: Done

File: lib/ui/profile/FarmDetails.dart (hunk 5 above)
Comment (davidfranquet) [reply lib/ui/profile/FarmDetails.dart]: Done

File: lib/ui/profile/FarmDetails.dart (hunk 3 above)
Comment (davidfranquet) [reply lib/ui/profile/FarmDetails.dart]: Done

**Comment by davidfranquet** [comment]
Added the general Alert widget on Delete profile action

**Review by davidfranquet — APPROVED** [review]

**Review by LeeMatthewHiggins — APPROVED** [review]

**Pull request merged**